pkg/agent/client.go

```diff
@@ -76,33 +85,154 @@ type AgentClient struct {
 
 	connManager *connectionManager
 
-	stream *RedialableAgentClient
-	stopCh <-chan struct{}
+	cs *ClientSet // the clientset that includes this AgentClient.
+
+	stream      agent.AgentService_ConnectClient
+	agentID     string
+	serverID    string // the id of the proxy server this client connects to.
+	serverCount int    // the number of the proxy server instances.
```

Comment:
Not sure we should keep serverCount here. It seems like there should only be one value for serverCount. Not sure what it means if we somehow get different clients with different values for serverCount. Seems better to just have serverCount in ClientSet.

Reply:
Done. Bubbled up the serverCount as a return value.